pkg/tls/marshal.go

```diff
@@ -103,10 +103,16 @@ func ParsePrivateKey(b []byte) (*rsa.PrivateKey, error) {
 		return nil, errors.New("extra data after decoding PEM block")
 	}
 
-	key, err := x509.ParsePKCS1PrivateKey(block.Bytes)
-	if err != nil {
-		return nil, err
+	if key, err := x509.ParsePKCS1PrivateKey(block.Bytes); err == nil {
```

Comment:
if you look at block.Type, you'll find it differs depending on whether it's a PKCS1 ("RSA PRIVATE KEY") or PKCS8 ("PRIVATE KEY").  Switch on that.